pettingzoo/test/parallel_test.py

```diff
@@ -80,3 +82,6 @@ def parallel_api_test(par_env, num_cycles=1000):
                     live_agents.remove(agent)
 
             assert set(par_env.agents) == live_agents
+
+            if len(live_agents) == 0:
```

Comment:
Honestly, not sure about this one. It fixes the test for `tiger_deer_v3` but line 24 fixes the test for the other one that had an error which was `adversarial_pursuit_v3` and that line makes sense whereas I feel like this one shouldn't be needed. For some reason tiger_deer keeps resetting at 100 and I cannot figure out why. It was doing this before but the difference is this PR no longer allows mismatching action settings to go through (see line 196 of `pettingzoo/magent/magent_env.py` above). Up to you if I should keep trying to figure tiger_deer out as part of this PR.

Reply:
Tiger-deer always resets at time step 100 because all the tigers die of starvation at the same time, because they failed to eat any of the deer.

Reply:
So I think this is intended behavior.

Reply:
Ah that's right. That reminds me, I was thinking about this case and whether this then makes for a good mechanism for determining if its done. The current implementation would make sense if we assume that in every use case, all agents being dead = done. If there are other conditions for doneness, maybe using something more direct is better. What do you think?